### ISSUE TITLE
Latest DB Update #26821 does not include EICAR signatures ?

### ISSUE THREAD

**Issue by reosch** [Cisco-Talos/clamav#846]
With the DB Update today, our Unit-Test Pipeline fails, since the EICAR Test File is not recognized as virus anymore.

Before the freshclam update it worked, after it fails, update log:

```
daily database available for update (local version: 26820, remote version: 26821)
Current database is 1 version behind.
Downloading database patch # 26821...
Time:    0.2s, ETA:    0.0s [========================>]   28.10KiB/28.10KiB
Testing database: '/var/lib/clamav/tmp.223eac5329/clamav-b103255f7cda7ea885fedfcc2ddc8ade.tmp-daily.cld' ...
Database test passed.
daily.cld updated (version: 26821, sigs: 2021707, f-level: 90, builder: raynman)
main.cvd database is up-to-date (version: 62, sigs: 6647427, f-level: 90, builder: sigmgr)
```

Testfile:
```
X5O!P%@AP[4\PZX54(P^)7CC)7}$EICAR-STANDARD-ANTIVIRUS-TEST-FILE!$H+H*

```

**Comment by reosch** [comment]
@micahsnyder some more information about our setup:

We use a docker container which is based on ubuntu 18.04 LTS

full update log which caused the problem
```
root@redacted-server:/var/www/redacted# freshclam
ClamAV update process started at Thu Feb 23 14:59:16 2023
WARNING: Your ClamAV installation is OUTDATED!
WARNING: Local version: 0.103.6 Recommended version: 0.103.8
DON'T PANIC! Read https://docs.clamav.net/manual/Installing.html
daily database available for update (local version: 26820, remote version: 26821)
Current database is 1 version behind.
Downloading database patch # 26821...
Time:    0.2s, ETA:    0.0s [========================>]   28.10KiB/28.10KiB
Testing database: '/var/lib/clamav/tmp.223eac5329/clamav-b103255f7cda7ea885fedfcc2ddc8ade.tmp-daily.cld' ...
Database test passed.
daily.cld updated (version: 26821, sigs: 2021707, f-level: 90, builder: raynman)
main.cvd database is up-to-date (version: 62, sigs: 6647427, f-level: 90, builder: sigmgr)
bytecode database available for update (local version: 333, remote version: 334)
Current database is 1 version behind.
Downloading database patch # 334...
Time:    0.2s, ETA:    0.0s [========================>]       801B/801B
Testing database: '/var/lib/clamav/tmp.223eac5329/clamav-0e3020b10697db7a057588a8ff7979c2.tmp-bytecode.cld' ...
Database test passed.
bytecode.cld updated (version: 334, sigs: 91, f-level: 90, builder: anvilleg)
Clamd successfully notified about the update.
```

clamconf -n
```
root@redacted-server:/var/www/redacted# clamconf -n
Checking configuration files in /etc/clamav

Config file: clamd.conf
-----------------------
LogFile = "/var/log/clamav/clamd.log"
LogTime = "yes"
ExtendedDetectionInfo = "yes"
PidFile = "/run/clamav/clamd.pid"
LocalSocket = "/run/clamav/clamd.sock"
TCPSocket = "3310"
StreamMaxLength = "52428800"
Foreground = "yes"
User = "clamav"
DetectPUA = "yes"

Config file: freshclam.conf
---------------------------
PidFile = "/run/clamav/freshclam.pid"
Foreground = "yes"
UpdateLogFile = "/var/log/clamav/freshclam.log"
DatabaseMirror = "database.clamav.net"

clamav-milter.conf not found

Software settings
-----------------
Version: 0.103.6
Optional features supported: MEMPOOL IPv6 FRESHCLAM_DNS_FIX AUTOIT_EA06 BZIP2 LIBXML2 PCRE ICONV JSON JIT

Database information
--------------------
Database directory: /var/lib/clamav
bytecode.cld: version 334, sigs: 91, built on Wed Feb 22 22:33:21 2023
daily.cld: version 26821, sigs: 2021707, built on Thu Feb 23 09:40:35 2023
main.cvd: version 62, sigs: 6647427, built on Thu Sep 16 14:32:42 2021
Total number of signatures: 8669225

Platform information
--------------------
uname: Linux 5.15.49-linuxkit #1 SMP PREEMPT Tue Sep 13 07:51:32 UTC 2022 x86_64
OS: linux-gnu, ARCH: x86_64, CPU: x86_64
Full OS version: Ubuntu 18.04.6 LTS
zlib version: 1.2.11 (1.2.11), compile flags: a9
Triple: x86_64-pc-linux-gnu
CPU: k8-sse3, Little-endian
platform id: 0x0a217f7f0807050001070500

Build information
-----------------
GNU C: 7.5.0 (7.5.0)
GNU C++: 7.5.0 (7.5.0)
CPPFLAGS: -Wdate-time -D_FORTIFY_SOURCE=2
CFLAGS: -g -O2 -fdebug-prefix-map=/build/clamav-6gS6Mc/clamav-0.103.6+dfsg=. -fstack-protector-strong -Wformat -Werror=format-security -Wall -D_FILE_OFFSET_BITS=64  -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64
CXXFLAGS: -g -O2 -fdebug-prefix-map=/build/clamav-6gS6Mc/clamav-0.103.6+dfsg=. -fstack-protector-strong -Wformat -Werror=format-security -Wall -D_FILE_OFFSET_BITS=64
LDFLAGS: -Wl,-Bsymbolic-functions -Wl,-z,relro -Wl,-z,now -Wl,--as-needed
Configure: '--build=x86_64-linux-gnu' '--prefix=/usr' '--includedir=/usr/include' '--mandir=/usr/share/man' '--infodir=/usr/share/info' '--sysconfdir=/etc' '--localstatedir=/var' '--disable-silent-rules' '--libdir=/usr/lib/x86_64-linux-gnu' '--libexecdir=/usr/lib/x86_64-linux-gnu' '--runstatedir=/run' '--disable-maintainer-mode' '--disable-dependency-tracking' 'CFLAGS=-g -O2 -fdebug-prefix-map=/build/clamav-6gS6Mc/clamav-0.103.6+dfsg=. -fstack-protector-strong -Wformat -Werror=format-security -Wall -D_FILE_OFFSET_BITS=64' 'CPPFLAGS=-Wdate-time -D_FORTIFY_SOURCE=2' 'CXXFLAGS=-g -O2 -fdebug-prefix-map=/build/clamav-6gS6Mc/clamav-0.103.6+dfsg=. -fstack-protector-strong -Wformat -Werror=format-security -Wall -D_FILE_OFFSET_BITS=64' 'LDFLAGS=-Wl,-Bsymbolic-functions -Wl,-z,relro -Wl,-z,now -Wl,--as-needed' '--with-dbdir=/var/lib/clamav' '--sysconfdir=/etc/clamav' '--disable-clamav' '--disable-unrar' '--enable-milter' '--enable-dns-fix' '--with-libjson' '--with-system-libmspack' '--with-libcurl=/usr' '--with-gnu-ld' '-with-system-llvm=/usr/bin/llvm-config-3.9' '--with-llvm-linking=dynamic' '--with-systemdsystemunitdir=/lib/systemd/system' 'build_alias=x86_64-linux-gnu' 'OBJCFLAGS=-g -O2 -fdebug-prefix-map=/build/clamav-6gS6Mc/clamav-0.103.6+dfsg=. -fstack-protector-strong -Wformat -Werror=format-security'
sizeof(void*) = 8
Engine flevel: 127, dconf: 127
```

clamav database directory
```
root@redacted-server:/var/lib/clamav# ls -lah
total 348M
drwxr-xr-x 1 clamav clamav 4.0K Feb 23 14:59 .
drwxr-xr-x 1 root   root   4.0K Jul  5  2022 ..
-rw-r--r-- 1 clamav clamav 1.4M Feb 23 14:59 bytecode.cld
-rw-r--r-- 1 clamav clamav 184M Feb 23 14:59 daily.cld
-rw-r--r-- 1 clamav clamav   69 Jul  5  2022 freshclam.dat
-rw-r--r-- 1 clamav clamav 163M Jul  5  2022 main.cvd
```

**Comment by micahsnyder** [comment]
@reosch thanks for the extra information. I tried the same thing with 0.103.8 and had no problem. 

```
❯ ~/clams/0.103.8/bin/freshclam
ClamAV update process started at Thu Feb 23 10:04:59 2023
daily database available for update (local version: 26693, remote version: 26821)
Current database is 128 versions behind.
Downloading database patch # 26694...
WARNING: downloadFile: file not found: https://database.clamav.net/daily-26694.cdiff
WARNING: downloadPatch: Can't download daily-26694.cdiff from https://database.clamav.net/daily-26694.cdiff
Downloading database patch # 26694...
WARNING: downloadFile: file not found: https://database.clamav.net/daily-26694.cdiff
WARNING: downloadPatch: Can't download daily-26694.cdiff from https://database.clamav.net/daily-26694.cdiff
Downloading database patch # 26694...
WARNING: downloadFile: file not found: https://database.clamav.net/daily-26694.cdiff
WARNING: downloadPatch: Can't download daily-26694.cdiff from https://database.clamav.net/daily-26694.cdiff
WARNING: Incremental update failed, trying to download daily.cvd
Time:    1.3s, ETA:    0.0s [========================>]   57.96MiB/57.96MiB
Testing database: '/home/micasnyd/clams/0.103.8/share/clamav/tmp.119b4a2434/clamav-9b33e8ff3d1ccf3c5e809923ff288ccd.tmp-daily.cvd' ...
Database test passed.
daily.cvd updated (version: 26821, sigs: 2021707, f-level: 90, builder: raynman)
main.cvd database is up-to-date (version: 62, sigs: 6647427, f-level: 90, builder: sigmgr)
bytecode database available for update (local version: 333, remote version: 334)
Current database is 1 version behind.
Downloading database patch # 334...
Time:    0.1s, ETA:    0.0s [========================>]       801B/801B
Testing database: '/home/micasnyd/clams/0.103.8/share/clamav/tmp.119b4a2434/clamav-78daec2fd4a50666a8074d85397d6d73.tmp-bytecode.cld' ...
Database test passed.
bytecode.cld updated (version: 334, sigs: 91, f-level: 90, builder: anvilleg)

❯ /home/micasnyd/clams/0.103.8/bin/clamscan ~/eicar.com.txt
/home/micasnyd/eicar.com.txt: Eicar-Signature FOUND

----------- SCAN SUMMARY -----------
Known viruses: 8653636
Engine version: 0.103.8
Scanned directories: 0
Scanned files: 1
Infected files: 1
Data scanned: 0.00 MB
Data read: 0.00 MB (ratio 0.00:1)
Time: 21.580 sec (0 m 21 s)
Start Date: 2023:02:23 10:05:30
End Date:   2023:02:23 10:05:52
```

I also just tested with 0.103.6 with the latest bytecode.cld file after an update and it worked okay.

By chance can you upload your `bytecode.cld` file for me to play with?

Afterwards, you may also try deleting the `bytecode.cld` file from your database directory and updating again.

**Comment by reosch** [comment]
Dear @micahsnyder I attached the [bytecode.cld.zip](https://github.com/Cisco-Talos/clamav/files/10822335/bytecode.cld.zip)

Deleting it and doing a freshclam + clamd restart results in the same problem.

This problem affects all our developer instances and pipelines, but they essentially use all the same container.

**I just reinstalled clamav by purging it and reinstalling via apt and it detects the eicar signature now. This is kinda odd to me, since it worked before and only failed after the freshclam update, reinstalling clamav solves it.** It's still the same version `ClamAV 0.103.6/26821/Thu Feb 23 09:40:35 2023`

I will see if I can play with it later, for now I will rebuild all our containers.
